### PR TITLE
Term update form ghost loading - jump to the errored field

### PR DESCRIPTION
**Pull request by G-Wang12**
Made sure that when a user enters an invalid value the form will automatically be scrolled to that invalid answer.

Clickup Task link and demo here: https://app.clickup.com/t/2mrab39

### CODE REVIEW

File: frontend/components/molecules/Nav.js [hunk 1]
```diff
@@ -51,39 +58,43 @@ const Nav = ({ navItems, index }) => {
...
+        </SystemComponent>
+        <SystemComponent
+          display="flex"
+          flexWrap="wrap"
+          justifyContent="flex-end"
+          mt={3}
+          width={['80%', '80%', '80%', 'auto']}
+        >
+          {navItems.map(({ name, link }, i) => (
+            <a href={link} key={i}>
```
Comment (Samanthatb1) [inline frontend/components/molecules/Nav.js:80]: Could you confirm that there's no warning in the browser console that says 
`Warning: Each child in an array should have a unique "key" prop.`

related [read](https://stackoverflow.com/questions/28329382/understanding-unique-keys-for-array-children-in-react-js/43892905#43892905)

**Review by Samanthatb1 — COMMENTED** [review]
Nice job. I'm still having TeamHub issues so I'm not able to run this branch locally yet. I've left some comments just based on the code

File: frontend/util.js [hunk 2]
```diff
@@ -54,6 +63,22 @@ export const isInvalidStudentId = (number) => {
     return !number.match(/^[0-9]*$/) || number.length > 8;
 }
 
+export const scrollToFirstError = (formErrors) => {
```
Comment (sashco-m) [inline frontend/util.js:66]: (I have very little FE knowledge) would something like this be useful https://www.npmjs.com/package/react-scroll-into-view

File: frontend/components/molecules/Nav.js [hunk 3]
```diff
@@ -51,39 +58,43 @@ const Nav = ({ navItems, index }) => {
...
         display="flex"
-        flexWrap="wrap"
-        justifyContent="flex-end"
-        mt={3}
-        width={['80%', '80%', '80%', 'auto']}
+        flexDirection="column"
+        alignItems="flex-end"
+        bg="greys.0"
+        shadow={scrolled ? 'default' : 'none'}
       >
```
[the remark this answers is not in the document]
Comment (SteveShengStar) [reply frontend/components/molecules/Nav.js]: It's okay, we are using Styled-System in addition to Styled-Components. In Styled-System, this is typically how you specify CSS styling.

But Gordon made changes here to make it into a Styled-Component component. So this should be good :)

File: frontend/components/molecules/Nav.js [hunk 4]
```diff
@@ -51,39 +58,43 @@ const Nav = ({ navItems, index }) => {
...
-        {navItems.map(({ name, link }, i) => (
-          <a href={link} key={i}>
-            <NavLink
-              fontSize={['smallNav', 'smallNav', 'smallNav', 'nav']}
-              fontWeight="bold"
```
[the remark this answers is not in the document]
Comment (SteveShengStar) [reply frontend/components/molecules/Nav.js:75]: Nope, we're good.

File: frontend/components/molecules/Nav.js (hunk 1 above)
Comment (SteveShengStar) [reply frontend/components/molecules/Nav.js:81]: I think Gordon fixed this now. He tested this locally to make sure there is no warning anymore. Thank you.

File: frontend/util.js (hunk 2 above)
Comment (SteveShengStar) [reply frontend/util.js:66]: Oh wow, neat feature. 

I think we can keep this method for now. But that's a cool package !

File: frontend/components/molecules/Nav.js [hunk 5]
```diff
@@ -7,19 +7,31 @@ import LogoutButton from './LogoutButton';
...
+  padding-right: ${(props) => props.theme.space[9]}px;
+  padding-top: ${(props) => props.theme.space[8]}px;
+  padding-bottom: ${(props) => props.theme.space[7]}px;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+  background-color: ${props => props.theme.colors.greys[0]};
+  shadow: ${props => props.scrolled ? 'default' : 'none'};
   ${(props) => props.theme.mediaQueries.tablet} {
-    right: ${(props) => props.theme.space.cardMargin}px;
```
Comment (LavanSumanan) [inline frontend/components/molecules/Nav.js:16]: is there a reason why you chose to hardcode a pixel size instead of using theme?

Comment (SteveShengStar) [reply frontend/components/molecules/Nav.js]: This change has been reverted.

**Pull request merged**